patches/api/0449-Add-more-scoreboard-API.patch

```diff
@@ -0,0 +1,86 @@
+Subject: [PATCH] Add more scoreboard API
+
+
+diff --git a/src/main/java/org/bukkit/scoreboard/Objective.java b/src/main/java/org/bukkit/scoreboard/Objective.java
+index a193ffabb05160b462dee1ba8f687fdbc84405b6..bd4d84cbf220ab02f09ece97873bbf0bdf7a45ba 100644
+--- a/src/main/java/org/bukkit/scoreboard/Objective.java
++++ b/src/main/java/org/bukkit/scoreboard/Objective.java
+@@ -175,4 +175,24 @@ public interface Objective {
+      */
+     @NotNull Score getScoreFor(@NotNull org.bukkit.entity.Entity entity) throws IllegalArgumentException, IllegalStateException;
+     // Paper end - improve scoreboard entries
++
++    // Paper start - add more score API
++    /**
++     * Gets if this objective will auto update score
++     * displays on changes.
++     *
++     * @return true if auto updating
++     * @throws IllegalStateException if this objective has been unregistered
++     */
++    boolean willAutoUpdateDisplay();
++
++    /**
++     * Sets if this objective will auto update
++     * score displays on changes.
++     *
++     * @param autoUpdateDisplay true to auto update
++     * @throws IllegalStateException if this objective has been unregistered
++     */
++    void setAutoUpdateDisplay(boolean autoUpdateDisplay);
++    // Paper end - add more score API
+ }
+diff --git a/src/main/java/org/bukkit/scoreboard/Score.java b/src/main/java/org/bukkit/scoreboard/Score.java
+index 1eaa9a93f8eff5f18a6cce2d74f21eb19db273c8..7fae0cb63aff6aa8bd2a4b6aeceab7aa64d8776f 100644
+--- a/src/main/java/org/bukkit/scoreboard/Score.java
++++ b/src/main/java/org/bukkit/scoreboard/Score.java
+@@ -83,4 +83,46 @@ public interface Score {
+      */
+     void resetScore() throws IllegalStateException;
+     // Paper end
++
++    // Paper start - add more score API
++    /**
++     * Gets if this score is locked and cannot
++     * be used by the {@code /trigger} command executed
++     * by the owner of this score.
++     *
++     * @return true if locked, false if unlocked
++     * @throws IllegalStateException if the score isn't set
++     * @throws IllegalStateException if the associated objective has been unregistered
++     * @see #isScoreSet()
++     */
++    boolean isLocked();
++
++    /**
++     * Sets if this score is locked and can
++     * be used by the {@code /trigger} command
++     * executed by the owner of this score.
++     *
++     * @param locked true to lock, false to unlock
++     * @throws IllegalStateException if the score isn't set
++     * @throws IllegalStateException if the associated objective has been unregistered
++     */
++    void setLocked(boolean locked);
++    /**
++     * Get the custom name for this entry.
++     *
++     * @return the custom name or null if not set
++     * @throws IllegalStateException if the score isn't set
++     * @throws IllegalStateException if the associated objective has been unregistered
++     */
++    @Nullable net.kyori.adventure.text.Component entryCustomName();
```

Comment:
The name is a bit weird, what about just `displayName` or `customName` to also stay consistent with other api?

Reply:
Went with "customName"